amr-wind/equation_systems/icns/icns_diffusion.H

```diff
@@ -293,25 +541,41 @@ struct DiffusionOp<ICNS, Scheme>
 {
     std::unique_ptr<ICNSDiffTensorOp> m_tensor_op;
     std::unique_ptr<ICNSDiffScalarOp> m_scalar_op;
+    std::unique_ptr<ICNSDiffScalarSegregatedOp> m_scalar_segregated_op;
```

Comment:
Just a question here. `ICNSDiffScalarOp` and `ICNSDiffScalarSegregatedOp` share a lot of code. Is there any way they can be consolidated by maybe sharing a base class? The additional advantage of that is that we can avoid many of the `if (use_segregated_op)` below by just instantiating with the right args.

Reply:
@marchdf 

Thank you for reviewing the PR. 

The segregated solve class `ICNSDiffScalarSegregatedOp` constructs and performs the solve operation for each of the momentum equation separately. On the other hand, `ICNSDiffScalarOp` constructs a tensor solve similar to what is done in `ICNSDiffTensorOp`. This is why m_solver_scalar is of type`amrex::Array<std::unique_ptr<amrex::MLABecLaplacian>, AMREX_SPACEDIM>`  in `ICNSDiffScalarSegregatedOp` and `std::unique_ptr<amrex::MLABecLaplacian>` in `ICNSDiffScalarOp`.

This PR is the first step towards incorporating the implicit time-stepping solver in AMR-Wind. In the next PR, we will be extending `ICNSDiffScalarSegregatedOp` to compute "C" coefficients in addition to "A" and "B". This is another reason for having a separate class for true segregated solve functionality in AMR-Wind.

Reply:
Ok cool, that makes total sense!